Kudu.Services/Diagnostics/CrashDumpController.cs

```diff
@@ -0,0 +1,121 @@
+﻿using Kudu.Contracts.Diagnostics;
+using Kudu.Contracts.Settings;
+using Kudu.Contracts.Tracing;
+using Kudu.Core;
+using Kudu.Core.Infrastructure;
+using Kudu.Core.Tracing;
+using Kudu.Services.Infrastructure;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.IO.Abstractions;
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using System.Threading.Tasks;
+using System.Web.Http;
+
+namespace Kudu.Services.Diagnostics
+{
+    public class CrashDumpController : ApiController
+    {
+        private readonly ITracer _tracer;
+        private readonly IEnvironment _environment;
+        private readonly IDeploymentSettingsManager _settings;
+
+        public CrashDumpController(ITracer tracer,
+                                 IEnvironment environment,
+                                 IDeploymentSettingsManager settings)
+        {
+            _tracer = tracer;
+            _environment = environment;
+            _settings = settings;
+        }
+
+        [HttpGet]
+        public IEnumerable<CrashDumpInfo> List()
+        {
+            using (_tracer.Step("CrashDumpController.List"))
+            {
+                return FileSystemHelpers.GetFiles(_environment.CrashDumpsPath, "*.dmp")
+                    .Select(FileSystemHelpers.FileInfoFromFileName)
+                    .Select(f => GetCrashDumpInfoFromFile(f, Request.RequestUri.GetLeftPart(UriPartial.Authority)));
+            }
+        }
+
+        [HttpGet]
+        public HttpResponseMessage Get(string name)
+        {
+            using (_tracer.Step($"CrashDumpController.Get({name})"))
+            {
+                return Request.CreateResponse(HttpStatusCode.OK, GetCrashDump(name));
+            }
+        }
+
+        [HttpDelete]
+        public HttpResponseMessage Delete(string name)
+        {
+            using (_tracer.Step($"CrashDumpController.Delete({name})"))
+            {
+                var crashDump = GetCrashDump(name);
+                FileSystemHelpers.DeleteFile(crashDump.FilePath);
+                return Request.CreateResponse(HttpStatusCode.OK);
+            }
+        }
+
+        [HttpPost]
+        public async Task<HttpResponseMessage> Analyze(string name)
+        {
+            using (_tracer.Step($"CrashDumpController.Analyze({name})"))
+            {
+                var crashDump = GetCrashDump(name);
+                var exe = new Executable(ResolveCdbPath(), _environment.RootPath, _settings.GetCommandIdleTimeout());
+                using (MemoryStream outputStream = new MemoryStream(), errorStream = new MemoryStream())
+                {
+                    // ExecuteAsync could deadlock if called from an ASP.NET thread
+                    var exitCode = await Task.Run(async () => await exe.ExecuteAsync(_tracer, $"-c \"!analyze -v;q\" -z \"{crashDump.FilePath}\"", outputStream, errorStream));
```

Comment:
you may also experiment `".symfix;.reload;.loadby sos clr;~*k;!analyze -v;q`.   If it still is not very useful, maybe not worth doing exposing `/analyze` at all.